Sources/CassandraClient/Data.swift

```diff
@@ -46,6 +48,35 @@ extension CassandraClient {
             Iterator(rows: self)
         }
 
+        /// Returns a reusable paging token.
+        ///
+        /// - Warning: This token is not suitable or safe for sharing externally.
+        public func opaquePagingStateToken() throws -> OpaquePagingStateToken {
+            try OpaquePagingStateToken(token: rawPagingStateToken())
+        }
+
+        private func rawPagingStateToken() throws -> [UInt8] {
+            var buffer: UnsafePointer<CChar>? = nil
+            var length: Int = 0
+
+            // The underlying memory is freed with the Rows result
+            let result = cass_result_paging_state_token(self.rawPointer, &buffer, &length)
+            guard result == CASS_OK, let bytesPointer = buffer else {
+                throw CassandraClient.Error(result)
+            }
+
+            let tokenBytes = bytesPointer.withMemoryRebound(to: UInt8.self, capacity: length) {
+                let bufferPointer = UnsafeBufferPointer(start: $0, count: length)
+                return Array(unsafeUninitializedCapacity: length) { storagePointer, storageCount in
+                    var (unwritten, endIndex) = storagePointer.initialize(from: bufferPointer)
+                    precondition(unwritten.next() == nil)
```

Comment:
do we use precondition elsewhere? otherwise maybe guard and throe an error so we dont crash the program?

Reply:
It looks like it is used quite extensively. I am always a bit apprehensive about using it for the reason you highlighted.

Someone suggested this in an earlier review. This isn't a normal error condition, so I don't think throwing is appropriate. For this to fail either means we passed differing lengths in to one of those calls (a really bad bug), or something has seriously gone wrong in the runtime internally.

As a guard against a programming error, it seems a little excessive, but were it to be a reality I think crashing is actually appropriate as things have totally gone off the rails.